### PR TITLE
fix: align ONNX-backed export arguments

### PR DESCRIPTION
**Pull request by amanharshx**
## Summary

Align export argument listings with ONNX-backed formats that already use `opset` and `simplify`.

## Motivation

Several formats use these options while creating an intermediate ONNX graph, but their argument tables and documentation did not list them. Accurate tables are needed before export validation can safely use them as its source of truth.

## Key Changes

- Add `opset` to MNN, SavedModel, GraphDef, Edge TPU, RKNN, DEEPX, and QNN.
- Add `simplify` to MNN, RKNN, DEEPX, and QNN.
- Keep options excluded where exporters override them.
- Sync integration pages, shared export table, tutorial notebook, and configuration wording.
- Deleted: outdated `ONNX/engine only` wording.

## 🛠️ PR Summary

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

### 🌟 Summary
Aligns ONNX-backed export arguments across supported Ultralytics export formats and their documentation.

### 📊 Key Changes
- Adds `opset` and, where supported, `simplify` to export format metadata for TensorFlow SavedModel, GraphDef, Edge TPU, MNN, RKNN, DEEPX, and Qualcomm QNN.
- Updates `default.yaml` descriptions to reflect support for intermediate ONNX graph simplification and opset selection.
- Synchronizes integration documentation, the export parameter table, and the tutorial notebook with the updated arguments.
- Documents RKNN's opset limit, which reduces values above 19 to 19.

### 🎯 Purpose & Impact
- Provides consistent argument handling for export formats that use an intermediate ONNX representation.
- Makes graph simplification and ONNX opset control visible and discoverable to users.
- Improves export compatibility and configurability for edge and deployment backends, with no changes to unrelated model workflows.

### CODE REVIEW

**Comment by UltralyticsAssistant** [comment]
👋 Hello @amanharshx, thank you for submitting a `ultralytics/ultralytics` 🚀 PR! This update aligns ONNX-backed export arguments across multiple deployment formats. To ensure a seamless integration, please review the following checklist. This is an automated message, and an engineer will assist with your PR. 🛠️

- ✅ **Define a Purpose**: Clearly explain the purpose of your fix or feature in your PR description, and link to any [relevant issues](https://github.com/ultralytics/ultralytics/issues). Ensure your commit messages are clear, concise, and adhere to the project's conventions.
- ✅ **Synchronize with Source**: Confirm your PR is synchronized with the `ultralytics/ultralytics` `main` branch. If it's behind, update it by clicking the 'Update branch' button or by running `git pull` and `git merge main` locally.
- ✅ **Ensure CI Checks Pass**: Verify all Ultralytics [Continuous Integration (CI)](https://docs.ultralytics.com/help/CI) checks are passing. If any checks fail, please address the issues.
- ✅ **Update Documentation**: Update the relevant [documentation](https://docs.ultralytics.com/) for any new or modified features.
- ✅ **Add Tests**: If applicable, include or update tests to cover your changes, and confirm that all tests are passing.
- ✅ **Sign the CLA**: Please ensure you have signed our [Contributor License Agreement](https://docs.ultralytics.com/help/CLA) if this is your first Ultralytics PR by writing "I have read the CLA Document and I sign the CLA" in a new message.
- ✅ **Minimize Changes**: Limit your changes to the **minimum** necessary for your bug fix or feature addition. _"It is not daily increase but daily decrease, hack away the unessential. The closer to the source, the less wastage there is."_ — Bruce Lee

For more guidance, please refer to our [Contributing Guide](https://docs.ultralytics.com/help/contributing). Don't hesitate to leave a comment if you have any questions. Thank you for contributing to Ultralytics! 🚀

File: ultralytics/engine/exporter.py [hunk 1]
```diff
@@ -186,21 +186,21 @@ def export_formats():
             "_saved_model",
             True,
             True,
-            ["batch", "data", "fraction", "quantize", "keras", "nms"],
+            ["batch", "data", "fraction", "quantize", "opset", "keras", "nms"],
```
Comment (UltralyticsAssistant) [inline ultralytics/engine/exporter.py:189]: 💡 **MEDIUM**: Adding these keys to `export_formats()` does not yet make the format table enforce `opset`/`simplify`. `validate_args()` only checks `dynamic`, `keras`, `nms`, `batch`, `fraction`, `data`, and `optimize`, so a user can still pass either option to formats that do not list it without an unsupported-argument error. Include `opset` and `simplify` in that validation set so these tables can serve as the intended source of truth.

**Review by UltralyticsAssistant — COMMENTED** [review]
## 🔍 PR Review

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

The exporter and documentation tables are otherwise consistently updated for the ONNX-backed formats. One validation gap remains: the newly listed arguments are not currently covered by format-specific argument validation.

💬 Posted 1 inline comment

File: ultralytics/engine/exporter.py (hunk 1 above)
Comment (amanharshx) [reply ultralytics/engine/exporter.py:189]: This PR stays focused on correcting the export tables.

**Comment by onuralpszr** [comment]
## ⚡ Actions Trigger

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

GitHub Actions below triggered via workflow dispatch for this PR at 2026-07-22 10:11:05 UTC with `@ultralytics/run-ci` command
(available commands are `@ultralytics/run-all`, `@ultralytics/run-ci`, and `@ultralytics/run-docker`):

* ✅ [CI](https://github.com/ultralytics/ultralytics/actions/runs/29910866743): `ci.yml` run 27150

**Review by onuralpszr — APPROVED** [review]
